### PR TITLE
os_stub/openssllib: Support more architectures

### PR DESCRIPTION
**Pull request by alistair23**

### CODE REVIEW

**Review by jyao1 — APPROVED** [review]

File: os_stub/cryptlib_openssl/CMakeLists.txt [hunk 1]
```diff
@@ -11,10 +11,22 @@ INCLUDE_DIRECTORIES(${LIBSPDM_DIR}/include
...
+elseif(ARCH STREQUAL "aarch64")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_AARCH64)
+elseif(ARCH STREQUAL "riscv32")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV32)
+elseif(ARCH STREQUAL "riscv64")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV64)
+elseif(ARCH STREQUAL "arc")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_ARC)
+else()
+    MESSAGE(FATAL_ERROR "Unkown ARCH")
```
Comment (steven-bellock) [inline os_stub/cryptlib_openssl/CMakeLists.txt:29]: `Unkown` -> `Unknown`

File: os_stub/openssllib/CMakeLists.txt [hunk 2]
```diff
@@ -12,10 +12,22 @@ if (TOOLCHAIN STREQUAL "CLANG")
...
+elseif(ARCH STREQUAL "aarch64")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_AARCH64)
+elseif(ARCH STREQUAL "riscv32")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV32)
+elseif(ARCH STREQUAL "riscv64")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV64)
+elseif(ARCH STREQUAL "arc")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_ARC)
+else()
+    MESSAGE(FATAL_ERROR "Unkown ARCH")
```
Comment (steven-bellock) [inline os_stub/openssllib/CMakeLists.txt:30]: `Unkown` -> `Unknown`

**Review by steven-bellock — CHANGES_REQUESTED** [review]
Does libspdm's instance of OpenSSL even work on `arc` / `arm`?

**Comment by alistair23** [comment]
I have dropped the `arc` and `arm`. I suspect they would work, but I can't test it

**Comment by steven-bellock** [comment]
For better or worse libspdm uses the UEFI profile of OpenSSL, which helps it fit into more limited systems.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**